### ISSUE TITLE
Remove the custom error rendering

### ISSUE THREAD

**Issue by n-b** [betagouv/conseillers-entreprises#1122]
See #1121 

TLDR: Basically, our custom error rendering (in SharedController#render_error) gets in the way of the normal Rails handling. We should get rid of it. (commit 3f61bff)

We're getting requests for sitemap.txt. The only supported type is xml, and a ActionController::UnknownFormat exception is raised. According to the docs (See also this relevant PR, this should result in a 406. However, we rescue in render_error: our handler can only return 404 ou 500, and only returns 404 for a very specific set of errors, so our 406 is changed to a 500.

In fact, this facility shouldn’t even exist. Our custom 404 and 500 pages were maybe meant to be customised at some point, but they are not. We should just use the default rails and use static 404.html etc. pages in /public. There's one catch with Sentry/Raven, as we still want to avoid reporting 404s to sentry, but it can be configured for it, see :excluded_exceptions and :should_capture. Maybe the default configuration just works™.

**Comment by n-b** [comment]
Turns out, it's not completely broken: when encountering a 500 or 404 error in the real world, the error partial is rendered within the layout of the controller that handled the query. That means, inside `layouts/application.html.haml` or `layouts/pages.html.haml`.

It’s slightly broken for `pages`, but it looks fine for `application`.
